app/services/badge/bestowment_service.rb

```diff
@@ -0,0 +1,55 @@
+# Handles creating and updating bestowments for a (badge, user) pair.
+class Badge
+  class BestowmentService
+    attr_reader :badge, :user
+
+    # @param [Class<Badge>] the badge to handle bestowments for
+    # @param [User] the user to bestow this badge upon
+    def initialize(badge, user)
+      @badge = badge
+      @user = user
+    end
+
+    # Actually creates/updates Bestowments for the badge
+    def run!
+      tracked_ranks.each do |(rank, progress)|
+        bestowment = bestowment_for(rank).first_or_initialize
+        # Ratcheting - progress can only increase, never decrease.
+        bestowment.progress = [bestowment.progres, progress].max
```

Comment:
Yeah, so it basically works like this:

```
like - 1
like - 2
like - 3
unlike - 3
like - 3
like - 4
```

So it never decreases, but it also won't increase until it reaches that point again.

This is mainly to prevent the progress from dropping on badges that have already earned, but I think it'd also annoy users to see their badge progress dropping 😛 

I could probably do this differently (just prevent earned badges from being modified) if @JoshFabian thinks a different logic would be better